coordinator/src/node/storage.rs

```diff
@@ -105,11 +109,60 @@ impl node::Storage for NodeStorage {
         outpoint: &OutPoint,
     ) -> Result<Option<SpendableOutputDescriptor>> {
         let mut conn = self.pool.get()?;
-        crate::db::spendable_outputs::get(&mut conn, outpoint)
+        db::spendable_outputs::get(&mut conn, outpoint)
     }
 
     fn all_spendable_outputs(&self) -> Result<Vec<SpendableOutputDescriptor>> {
         let mut conn = self.pool.get()?;
-        crate::db::spendable_outputs::get_all(&mut conn)
+        db::spendable_outputs::get_all(&mut conn)
+    }
+
+    // Channel
+
+    fn upsert_channel(&self, channel: Channel) -> Result<()> {
```

Comment:
I suppose because of the stupid generic `event_handler`.

I already documented the friction in the relevant issue: https://github.com/get10101/10101/issues/885.

Reply:
The shared event handler is the main reason.

However this data is also useful to the app side. Especially remembering broadcasted transactions.

@luckysori could that eventually replace remembering the spendable outputs?

Reply:
> could that eventually replace remembering the spendable outputs?

I would still trust LDK to generate the spendable-output events rather than try to be smarter by getting ahead of it by considering these transactions instead. I think we still need some extra information in the event to actually spend the output straight away.

Reply:
> I think we still need some extra information in the event to actually spend the output straight away.

The current implementation of the transaction does not save enough data true, but we could extend it.

Aren't we storing the spendable outputs because a broadcasted transaction failed? Or would we run into a risk of losing the transaction after all when the app dies before broadcasting?

Reply:
We store them unconditionally, because you could broadcast a transaction successfully and it might still never get confirmed.

Handling such scenarios is expected to be done with #846.